mico-core/src/main/java/io/github/ust/mico/core/service/MicoStatusService.java

```diff
@@ -79,46 +120,59 @@ public MicoApplicationStatusDTO getApplicationStatus(MicoApplication micoApplica
         try {
             deploymentOptional = micoKubernetesClient.getDeploymentOfMicoService(micoService);
         } catch (KubernetesResourceException e) {
             log.error("Error while retrieving Kubernetes deployment of MicoService '{}' '{}'. Continue with next one. Caused by: {}",
                 micoService.getShortName(), micoService.getVersion(), e.getMessage());
         }
-        if (!deploymentOptional.isPresent()) {
+        if (deploymentOptional.isPresent()) {
+            Deployment deployment = deploymentOptional.get();
+            serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
+            serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
+        } else {
             log.warn("There is no deployment of the MicoService '{}' '{}'. Continue with next one.",
                 micoService.getShortName(), micoService.getVersion());
+            return serviceStatus.setErrorMessages(CollectionUtils.listOf("No deployment of " + micoService.getShortName() + " " + micoService.getVersion() + " is available."));
         }
-
-        Deployment deployment = deploymentOptional.get();
-        MicoServiceStatusDTO serviceStatus = new MicoServiceStatusDTO();
         serviceStatus.setName(micoService.getName());
         serviceStatus.setShortName(micoService.getShortName());
         serviceStatus.setVersion(micoService.getVersion());
-        serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
-        serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
 
-        // Get status information for service interfaces of a serivce
+        // Get status information for service interfaces of a service
         List<MicoServiceInterfaceDTO> micoServiceInterfaceDTOList = getServiceInterfaceStatus(micoService);
         serviceStatus.setInterfacesInformation(micoServiceInterfaceDTOList);
 
+        // Return the names of all applications that are using this service
+        List<MicoApplication> usingApplications = micoApplicationRepository.findAllByUsedService(micoService.getShortName(), micoService.getVersion());
+        for (MicoApplication micoApplication: usingApplications) {
+            MicoUsingApplicationDTO usingApplication = new MicoUsingApplicationDTO(micoApplication.getName(), micoApplication.getShortName(), micoApplication.getVersion());
+            serviceStatus.getUsingApplications().add(usingApplication);
+        }
+
         // Get status information for all pods of a service
+        int averageCpuLoad = 0;
+        int averageMemoryUsage = 0;
         List<Pod> podList = micoKubernetesClient.getPodsCreatedByDeploymentOfMicoService(micoService);
         List<KubernetesPodInfoDTO> podInfos = new LinkedList<>();
         for (Pod pod : podList) {
             KubernetesPodInfoDTO podInfo = getUiPodInfo(pod);
+            averageCpuLoad += podInfo.getMetrics().getCpuLoad();
```

Comment:
Variable naming is misleading, `averageCpuLoad`actually is the total CPU load.

Reply:
Same for memory load.

Reply:
What if the pods are running on different nodes?

Reply:
Renamed variables to sumCpuLoad and sumMemoryUsage. The average CPU load and the average memory usage is now calculated for all pods that are running on the same node. The result is a list with all nodes and the associated, calculated values for the average CPU load and the average memory usage.